src/main/java/com/example/vaccinecalendar/controller/CalendarController.java

```diff
@@ -16,26 +17,32 @@
 public class CalendarController {
     private final ChildService childService;
 
-    @GetMapping("/new")
-    public String getNewCalendar(){
+    @GetMapping("/new/{id}")
+    public String getNewCalendar(@PathVariable("id") Long id, Model model){
+        ChildModel child = childService.getChildById(id);
+        model.addAttribute("child", child);
```

Comment:
Dużo duplikacji kodu, ja bym to wyodrębnił. Chyba że to tylko tymczasowo, to wtedy OK.